src/components/VideoList/VideoList/VideoList.spec.jsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+
+import { BrowserRouter } from 'react-router-dom';
+import DarkProvider from '../../../providers/DarkMode';
+import ThemesProvider from '../../../providers/Theme';
+
+import VideoList from './VideoList.component';
+
+import useVideos from '../../../utils/hooks/useVideos';
+
+import { REQUEST_API_TYPES } from '../../../utils/constants';
+
+jest.mock('../../../utils/hooks/useVideos');
+
+describe('Testing VideoList', () => {
+  const REQ_TYPE = REQUEST_API_TYPES.SEARCH_BY_TERM;
+
+  describe('Testing CardList rendering', () => {
+    it('Should display loading message, when isLoading is true', () => {
+      useVideos.mockReturnValueOnce({ videoList: null, isLoading: true });
+
+      const { container } = render(<VideoList />);
+
+      expect(container).toHaveTextContent(/loading/i);
+    });
+
+    it('Should return null, when videoList is null and isLoading is false', () => {
+      useVideos.mockReturnValueOnce({ videoList: null, isLoading: false });
+
+      const { container } = render(<VideoList />);
+
+      expect(container).toBeEmptyDOMElement();
+    });
+
+    it('should display "No Video Found" when result is empty array and isLoading is false', async () => {
+      useVideos.mockReturnValueOnce({ videoList: [], isLoading: false });
+
+      const { container } = render(<VideoList />);
+
+      expect(container).toHaveTextContent(/no video found/i);
+    });
+
+    it('should display video Cards when array is returned', async () => {
+      const returnedArray = [
```

Comment:
Since this is a quite large mock, I recommend you to move it to a separated `__mocks__/youtubeResponse.js` file (suggested name), so that your tests are cleaner.